cmake/ext_project_autotools.cmake

```diff
@@ -0,0 +1,118 @@
+        ARGS "${options}" "${sv_keywords}" "${mv_keywords}" ${ARGN}
+    )
+    if (NOT DEFINED ARGS_DIR)
+        message(FATAL_ERROR "DIR argument is mandatory")
+    endif()
+
+    # CONFIGURE_DEPENDS available since 3.12
+    # VERSION_GREATER_EQUAL is only since 3.7
+    if (${CMAKE_VERSION} VERSION_GREATER "3.12" OR
+        ${CMAKE_VERSION} VERSION_EQUAL "3.12")
+        set(autotool_files_standard
+            configure.ac
+            acinclude.m4
+        )
+        list_add_prefix(
+            autotool_files_standard "${ARGS_DIR}/" autotool_files_standard
+        )
+
+        file(
+            GLOB_RECURSE autotool_files_am
+            CONFIGURE_DEPENDS
+            RELATIVE "${CMAKE_SOURCE_DIR}"
+            "${ARGS_DIR}/*.am"
+        )
+
+        set(autotool_files ${autotool_files_am} ${autotool_files_standard})
+
+        file(
+            GLOB_RECURSE config_files_in
+            CONFIGURE_DEPENDS
+            RELATIVE "${CMAKE_SOURCE_DIR}"
+            "${ARGS_DIR}/*.in"
+        )
+
+        # Makefile is used as dependency between configure and make thus we
+        # need to exclude it from dependency for config.status step as
+        # it cannot be output of two custom commands.
+        list(REMOVE_ITEM config_files_in ${ARGS_DIR}/Makefile.in)
+    endif()
+
+    list_add_prefix(ARGS_BYPRODUCTS "${ARGS_DIR}/" byproducts)
+
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/${ARGS_DIR})
+
+    add_custom_command(
+        OUTPUT ${CMAKE_SOURCE_DIR}/${ARGS_DIR}/configure
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/${ARGS_DIR}
+        COMMAND
+            autoreconf -i
+        COMMAND
+            # autoreconf do not touch configure on all paths. For example if
+            # only *.am are changed. We don't aim to follow precise dependency
+            # path of autotools thus let's just update configure in this case.
+            touch configure
+        DEPENDS ${autotool_files}
+    )
+
+    add_custom_command(
+        OUTPUT ${ARGS_DIR}/Makefile
+        WORKING_DIRECTORY ${ARGS_DIR}
+        COMMAND ${CMAKE_SOURCE_DIR}/${ARGS_DIR}/configure ${ARGS_CONFIGURE}
+        DEPENDS ${ARGS_DIR}/configure
+    )
+
+    # Use artificial timestamp file to make a dependency with main target.
+    # We can't use just ${config_files_in} counterparts because
+    # depending on configure options not all counterparts could be generated
+    # by config.status and as a result it will be rerun unconditionally.
+    add_custom_command(
+        OUTPUT
+            ${ARGS_DIR}/config.stamp.tarantool
+        WORKING_DIRECTORY ${ARGS_DIR}
+        COMMAND ./config.status
+        COMMAND touch config.stamp.tarantool
+        DEPENDS
+            ${config_files_in}
+            # Actually we depend on config.status but it updates
+            # it's own timestamp on every run.
+            ${ARGS_DIR}/Makefile
+    )
+
+    add_custom_target(${name}
+        WORKING_DIRECTORY ${ARGS_DIR}
+        COMMAND
+            make -j
```

Comment:
It is undesirable to run unlimited number of make jobs: the build may fail due to OOM on a machine with low amount of memory. See #7020 for example. See also https://github.com/tarantool/tarantool/pull/7354#discussion_r914740882.

Reply:
Maybe [CMAKE_BUILD_PARALLEL_LEVEL](https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html) should do the trick, but it is available since CMake 3.12.

Reply:
Looks like better option for getting cpu number is [ProcessorCount](https://cmake.org/cmake/help/latest/module/ProcessorCount.html) module:

```
--- a/cmake/ext_project_autotools.cmake
+++ b/cmake/ext_project_autotools.cmake
@@ -1,4 +1,5 @@
 include(utils)
+include(ProcessorCount)

 # Build a third-party project that uses autotools to generate build system.
 # Basically it generates configure script, run configure and then make and
@@ -105,10 +106,17 @@ function(ext_project_autotools name)
             ${ARGS_DIR}/Makefile
     )

+    ProcessorCount(nproc)
+    if (nproc)
+        set(make make -j${nproc})
+    else()
+        set(make make)
+    endif()
+
     add_custom_target(${name}
         WORKING_DIRECTORY ${ARGS_DIR}
         COMMAND
-            make -j
+            ${make}
         DEPENDS
             ${ARGS_DIR}/Makefile
             ${ARGS_DIR}/config.stamp.tarantool
```

Reply:
Nice!